argilla/src/argilla/_helpers/_deploy.py

```diff
@@ -0,0 +1,195 @@
+
+_SLEEP_TIME = 10
+_FROM_REPO_ID = "argilla/argilla-template-space"
+
+
+class SpacesDeploymentMixin(LoggingMixin):
+    @classmethod
+    def deploy_on_spaces(
+        cls,
+        api_key: str,
+        repo_name: Optional[str] = "argilla",
+        org_name: Optional[str] = None,
+        hf_token: Optional[str] = None,
+        space_storage: Optional[Union[str, "SpaceStorage", None]] = None,
+        space_hardware: Optional[Union[str, "SpaceHardware"]] = "cpu-basic",
+        private: Optional[Union[bool, None]] = False,
+        overwrite: Optional[Union[bool, None]] = False,
+    ) -> "Argilla":
+        """
+        Deploys Argilla on Hugging Face Spaces.
+
+        Args:
+            api_key (str): The Argilla API key to be defined for the owner user and creator of the Space.
+            repo_name (Optional[str]): The ID of the repository where Argilla will be deployed. Defaults to "argilla".
+            org_name (Optional[str]): The name of the organization where Argilla will be deployed. Defaults to None.
+            hf_token (Optional[Union[str, SpaceStorage, None]]): The Hugging Face authentication token. Defaults to None.
+            space_storage (Optional[Union[str, SpaceStorage, None]]): The persistant storage size for the space. Defaults to None without persistant storage.
+            space_hardware (Optional[Union[str, SpaceStorage, None]]): The hardware configuration for the space. Defaults to "cpu-basic" with downtime after 48 hours of inactivity.
+            private (Optional[Union[bool, None]]): Whether the space should be private. Defaults to False.
+            overwrite (Optional[Union[bool, None]]): Whether to overwrite the existing space. Defaults to False.
+
+        Returns:
+            Argilla: The Argilla client.
+
+        Example:
+            ```Python
+            import argilla as rg
+
+            client = rg.Argilla.deploy_on_spaces(api_key="12345678")
+            ```
+        """
+        hf_token = cls._acquire_hf_token(ht_token=hf_token)
+        api = HfApi(token=hf_token)
+
+        # Get the org name from the repo name or default to the current user
+        token_username = api.whoami(token=hf_token)["name"]
+        org_name = org_name or token_username
+        repo_id = f"{org_name}/{repo_name}"
+
+        # Define the api_key for the space
+        secrets = [
+            {"key": "API_KEY", "value": api_key, "description": "The API key of the owner user."},
+            {"key": "USERNAME", "value": token_username, "description": "The username of the owner user."},
+            {"key": "WORKSPACE", "value": "argilla", "description": "The workspace of the space."},
+        ]
+
+        # Check if the space already exists
+        if api.repo_exists(repo_id=repo_id, repo_type="space", token=hf_token):
+            if cls._check_if_runtime_can_be_build(api.get_space_runtime(repo_id=repo_id, token=hf_token)):
+                api.restart_space(repo_id=repo_id, token=hf_token)
```

Comment:
It checks if the space can be restarted, i.e. is in a valid state without errors and without it being in the process of being built. If it can, the space is just restarted/rebooted, to make the code more reusable.